src/main/java/edu/byu/cs/autograder/QualityAnalyzer.java

```diff
@@ -39,17 +41,17 @@ public QualityOutput runQualityChecks(File stageRepo) {
         output = output.replaceAll(stageRepo.getAbsolutePath(), "");
         output = output.replaceAll(stageRepo.getPath(), "");
 
-        QualityAnalysis analysis = parseAnalysis(output);
-        float score = getScore(analysis);
-        String results = getResults(analysis);
-        String notes = getNotes(analysis);
-        return new QualityOutput(score, results, notes);
+        parseOutput(output);
+        float score = getScore();
+        String results = getResults();
+        String notes = getNotes();
+        return new QualityAnalysis(score, results, notes);
     }
 
 
-    private QualityAnalysis parseAnalysis(String output) {
-        Map<String, List<String>> errors = new HashMap<>();
-        List<String> warnings = new ArrayList<>();
+    private void parseOutput(String output) {
+        errors = new HashMap<>();
```

Comment:
Some people will argue that the code is cleaner when each method is architected as a [pure function](https://en.wikipedia.org/wiki/Pure_function), rather than relying on and mutating local variables.

In this case, it would seem straight forward to have this `parseOutput()` method return the `HashMap<> errors` and the `ArrayList<> warnings`, maybe as a record or a pair, and then explicitly pass them in to the subsequent methods that depend on them. The main advantage of this approach is that each method individually behaves the same given the same input every time. As a general rule, it's safer to follow acknowledging that this code seems to run just fine as it is.